core/src/main/java/org/infinispan/xsite/GlobalXSiteAdminOperations.java

```diff
@@ -80,6 +85,34 @@ public final String cancelPushState(@Parameter(description = "The destination si
       return performOperation(operations -> operations.cancelPushState(site));
    }
 
+   public final Map<String, SiteStatus> globalStatus() {
+      final Iterator<CacheXSiteAdminOperation> iterator = collectXSiteAdminOperation().iterator();
+      if (!iterator.hasNext()) {
+         return Collections.emptyMap();
+      }
+      Map<String, ContainerSiteStatusBuilder> siteStatusBuilderMap = new HashMap<>();
+      while (iterator.hasNext()) {
+         CacheXSiteAdminOperation xsiteAdminOperation = iterator.next();
+         Map<String, SiteStatus> perCacheSiteStatus = xsiteAdminOperation.xSiteAdminOperations.clusterStatus();
+         for (Map.Entry<String, SiteStatus> siteStatus : perCacheSiteStatus.entrySet()) {
```

Comment:
perCacheSiteStatus.forEach((siteName, status) -> { ... }); would be nicer to read and more compact